src/items/services/history.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@nestjs/common';
+import { AccountRepository } from '../../account/repositories/account.repository';
+import { NotFoundException } from '../../common/exceptions/exception-types';
+import { History } from '../entities/history.entity';
+import { TransactionType } from '../enums/transaction-type.enum';
+import { HistoryRepository } from '../repositories/history.repository';
+import { ItemRepository } from '../repositories/item.repository';
+
+@Injectable()
+export class HistoryService {
+  constructor(
+    private itemRepository: ItemRepository,
+    private historyRepository: HistoryRepository,
+    private accountRepository: AccountRepository
+  ) {}
+
+  async findAllByItemId(itemId: string): Promise<History[]> {
+    return this.historyRepository.findAllByItemId(itemId);
+  }
+
+  async createHistory(historyRequestDto: {
+    itemId: string;
```

Comment:
Take advantage of the HistoryRequestDto object you've already created instead of creating an anonymous object with the same shape, the code will look cleaner